packages/office-ui-fabric-react/src/components/MessageBar/examples/MessageBar.Basic.Example.tsx

```diff
@@ -1,160 +1,184 @@
-            <MessageBarButton>Yes</MessageBarButton>
-            <MessageBarButton>No</MessageBarButton>
-          </div>
-        }
-      >
-        Warning lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi luctus, purus a lobortis tristique, odio augue pharetra
-        metus, ac placerat nunc mi nec dui. Vestibulum aliquam et nunc semper scelerisque. Curabitur vitae orci nec quam condimentum
-        porttitor et sed lacus. Vivamus ac efficitur leo. Cras faucibus mauris libero, ac placerat erat euismod et. Donec pulvinar commodo
-        odio sit amet faucibus. In hac habitasse platea dictumst. Duis eu ante commodo, condimentum nibh pellentesque, laoreet enim. Fusce
-        massa lorem, ultrices eu mi a, fermentum suscipit magna. Integer porta purus pulvinar, hendrerit felis eget, condimentum mauris.
-        <Link href="www.bing.com" target="_blank">
-          Visit our website.
-        </Link>
-      </MessageBar>
-    </StackItem>
+  const WarningExample2 = () => (
+    <MessageBar
+      onDismiss={resetChoice}
+      dismissButtonAriaLabel="Close"
+      messageBarType={MessageBarType.warning}
+      ariaLabel="Aria help text here"
+      actions={
+        <div>
+          <MessageBarButton>Yes</MessageBarButton>
+          <MessageBarButton>No</MessageBarButton>
+        </div>
+      }
+    >
+      Warning lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi luctus, purus a lobortis tristique, odio augue pharetra metus,
+      ac placerat nunc mi nec dui. Vestibulum aliquam et nunc semper scelerisque. Curabitur vitae orci nec quam condimentum porttitor et sed
+      lacus. Vivamus ac efficitur leo. Cras faucibus mauris libero, ac placerat erat euismod et. Donec pulvinar commodo odio sit amet
+      faucibus. In hac habitasse platea dictumst. Duis eu ante commodo, condimentum nibh pellentesque, laoreet enim. Fusce massa lorem,
+      ultrices eu mi a, fermentum suscipit magna. Integer porta purus pulvinar, hendrerit felis eget, condimentum mauris.
+      <Link href="www.bing.com" target="_blank">
+        Visit our website.
+      </Link>
+    </MessageBar>
+  );
 
-    <StackItem>
-      <Label>Warning MessageBar - single line, with dismiss, action buttons, and custom icons</Label>
-      <MessageBar
-        messageBarType={MessageBarType.warning}
-        isMultiline={false}
-        onDismiss={log('test')}
-        dismissButtonAriaLabel="Close"
-        actions={
-          <div>
-            <MessageBarButton>Action</MessageBarButton>
-          </div>
-        }
-        dismissIconProps={{ iconName: 'AddFriend' }}
-        messageBarIconProps={{ iconName: 'Search' }}
-      >
-        Warning lorem ipsum dolor sit amet, a elit sem interdum consectetur adipiscing elit.{' '}
-        <Link href="www.bing.com" target="_blank">
-          Visit our website.
-        </Link>
-      </MessageBar>
-    </StackItem>
-  </Stack>
-);
+  return (
+    <Stack horizontal tokens={{ childrenGap: 16 }}>
```

Comment:
Recently we've been trying to move towards avoiding inline objects or functions since they mutate every render and break memoization if it's being used. (Currently most of our components aren't memoized, but I think the idea was to demonstrate best practices to people who may be copy-pasting our examples.) 

Usually for any constant objects or functions (don't depend any variables), I've been moving them somewhere outside the component. For the ones that depend on state or something you can use `useCallback` or `useMemo`.

Reply:
addressed